packages/@lwc/engine-core/src/framework/modules/attr-unless-prop.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import {
+    htmlAttributeToProperty,
+    isNull,
+    isUndefined,
+    StringCharCodeAt,
+    XML_NAMESPACE,
+    XLINK_NAMESPACE,
+} from '@lwc/shared';
+
+import { unlockAttribute, lockAttribute } from '../attributes';
+import { RendererAPI } from '../renderer';
+import { EmptyObject } from '../utils';
+import { VBaseElement } from '../vnodes';
+
+const ColonCharCode = 58;
+
+export function patchAttrUnlessProp(
+    oldVnode: VBaseElement | null,
+    vnode: VBaseElement,
+    renderer: RendererAPI
+) {
+    const {
+        data: { attrs },
+        elm,
+    } = vnode;
+    if (isUndefined(elm) || isUndefined(attrs)) {
```

Comment:
when will `elm` be undefined here? this is a vnode that is being patched... the elm should always exist at this point.